trace.go

```diff
@@ -108,6 +108,20 @@ func removeObserverFrom(parent **spanObserverTuple, ref *spanObserverTuple) (
 // Id returns the id of the Trace
 func (t *Trace) Id() int64 { return t.id }
 
+// GetAll returns values associated with a trace. See SetAll.
+func (t *Trace) GetAll() (val map[interface{}]interface{}) {
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+	return t.vals
+}
+
+// SetAll replace all key/value on a trace with a new sets of key/value.
+func (t *Trace) SetAll(vals map[interface{}]interface{}) {
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+	t.vals = vals
```

Comment:
Arguably, this should also make a copy to prevent modifications to the map applying to the trace. 

I don't want to make two copies, though, so perhaps a better API would be like 

```go
func (t *Trace) CopyFrom(s *Trace)
```

and maybe even unexported so we don't have to worry about API misuse or whatever.

Reply:
made it unexported